pkg1/elastic/elasticsearch.go

```
@@ -0,0 +1,190 @@
+		fmt.Println(err)
+	}
+	cfg := elasticsearch.Config{
+		Addresses: []string{
+			*esAddress,
+		},
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true,
+				Certificates: []tls.Certificate{cert}},
+		},
+	}
+	esClient, err := elasticsearch.NewClient(cfg)
+	if err != nil {
+		fmt.Println("Error", err)
+	}
+
+	return esClient
+}
+
+func NewElasticRepository() logs.LogsProvider {
+	esClient := InitializeElasticSearchClient()
+	repository := &ElasticRepository{esClient: esClient}
+	return repository
+}
+
+func (repository *ElasticRepository) FilterByIndex(index string) []string {
+	esClient := repository.esClient
+	searchResult, err := esClient.Search(
+		esClient.Search.WithContext(context.Background()),
+		esClient.Search.WithIndex(index),
+		esClient.Search.WithTrackTotalHits(true),
+		esClient.Search.WithPretty(),
+	)
+	if err != nil {
+		fmt.Println("Error getting response: %s", err)
+	}
+
+	var logs []string // create a slice of type string to append logs to
+
+	logs = getRelevantLogs(searchResult)
+
+	return logs
+
+}
+func (repository *ElasticRepository) FilterByTime(startTime string, finishTime string) []string {
+
+	fmt.Println(startTime)
+	fmt.Println(finishTime)
+
+	var logs []string // create a slice of type string to append logs to
+	//
+	esClient := repository.esClient
+	query := fmt.Sprintf(`{
+		"query": {
+		"range" : {
+			"@timestamp" : {
+				"gte": "%s",
+  				"lte": "%s",
+ 				"time_zone": "+00:00"
+			}
+		}
+	}
+	}`, startTime, finishTime)
+
+	var b strings.Builder
+	b.WriteString(query)
+	body := strings.NewReader(b.String())
+	searchResult, err := esClient.Search(
+		esClient.Search.WithContext(context.Background()),
+		esClient.Search.WithIndex(constants.InfraIndexName, constants.AuditIndexName, constants.AppIndexName),
+		esClient.Search.WithBody(body),
+		esClient.Search.WithTrackTotalHits(true),
+		esClient.Search.WithPretty(),
+	)
+	if err != nil {
+		fmt.Println("Error getting response: %s", err)
```

If an error happens in the application logic layer, it makes sense to notify a client about it. In this case, the error is ignored by the API, and a client won't get any error or results. Also, it is a potential place for `nil pointer dereference` as you try to access `searchResult` even if an error happens.